sql/sql_acl.cc

```diff
@@ -14312,6 +14313,50 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else
+  if (redirect_enabled)
```

Comment:
this looks like its going to redirect on any error. Is this the intended case?

Reply:
It won't redirect on error. This is the place where the incoming request passes all error checks and server is ready to send the OK packet.